containerbuddy/config.go

```diff
@@ -311,8 +313,19 @@ func initializeConfig(config *Config) (*Config, error) {
 			return nil, ifaceErr
 		}
 
-		if service.ipAddress, err = GetIP(interfaces); err != nil {
-			return nil, err
+		if service.Endpoint == "" && service.ipAddress ==  "" {
+			if service.ipAddress, err = GetIP(interfaces); err != nil {
+				return nil, err
+			}
+		} else {
+			if ok := net.ParseIP(service.Endpoint); ok == nil {
+				if ip, err := net.LookupHost(service.Endpoint); err == nil {
```

Comment:
I have concerns about doing name resolution in Containerbuddy. It's actually a very significant expansion of scope for the code.

Name resolution is hardly new technology, but the failure modes are often misunderstood or underestimated. Granted, my concerns stem from some deep work I did recently on CNS (https://www.joyent.com/blog/introducing-triton-container-name-service). Among those concerns:

1. The DNS TTL here cannot be honored
1. If we try to honor the TTL, how?
1. Multiple IPs may be returned for this A or AAA record, is it safe to assume only the first one should be used?